bundle/edu.gemini.spModel.io/src/main/scala/edu/gemini/spModel/io/impl/migration/to2015B/BrightnessParser.scala

```diff
@@ -19,35 +19,49 @@ object BrightnessParser extends RegexParsers {
     """\s*[~=,]?\s*""".r ^^^ { () }
 
   private val at: Parser[Unit] =
-    """\s*(nucleus at|at|@|/|:)?\s*""".r ^^^ { () }
+    """\s*(nucleus at|at|in|@|/|:)?\s*""".r ^^^ { () }
 
-  private val num: Parser[Double] =
-    """-?\d+(\.\d*)?""".r <~ """\s*(mag)?""".r ^^ (_.toDouble)
+  private val num: Parser[BigDecimal] =
+    """-?\d+(\.\d*)?""".r <~ """\s*(mag)?""".r ^^ (s => BigDecimal(new java.math.BigDecimal(s))) // BigDecimal.exact(s)
 
   private val band: Parser[Band] = {
     def bandParser(b: Band): Parser[Band] = {
-      val p = b.toString.r <~ """\s*(_mag|mag|-band|band)?""".r ^^^ b
+      val p = (b.toString + """'?\s*(_mag|mag|-band|band)?""").r ^^^ b
       p | paren(p) | paren(p, '<', '>')
     }
-    Band.values.map(bandParser).foldRight[Parser[Band]](failure("expected band"))(_ | _)
+
+    Band.values.map(bandParser).foldRight[Parser[Band]](failure("expected band"))(_ ||| _)
```

Comment:
Apparently, though it would be worth playing with "reluctant" versions of the regex.  The issue seems to be partially matching the U band.  If you put UC first in the band list it also works.